packages/react-reconciler/src/__tests__/ReactUse-test.js

```diff
@@ -1915,4 +1917,39 @@ describe('ReactUse', () => {
     assertLog(['Hi', 'World']);
     expect(root).toMatchRenderedOutput(<div>Hi World</div>);
   });
+
+  it('regression: does not get stuck in pending state after `use` suspends', async () => {
+    // This is a regression test. The root cause was an issue where we failed to
+    // switch from the "re-render" dispatcher back to the "update" dispatcher
+    // after a `use` suspends and triggers a replay.
+    let update;
+    function App({promise}) {
+      useState(false);
```

Comment:
Is this testing that everything **before** `use` is still processed appropriately? May be worth a comment because right now it looks like we could remove it.

Reply:
I was testing both and forgot to delete, I'll add another test for this one